### PR TITLE
Feat/1560 xms login button

### LINKED ISSUE #1560: As a Data Analyst user, I need to have a frontend button to direct me to XMS login portal
**Description:**
_Provide a brief background and justification for this issue_
We will need to update our current Login.gov button to a NextGenXMS button.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] Splash page will only have two buttons: one for login.gov via XMS and one for ACF AMS
- [ ] New button links to XMS portal
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] Update button and link to XMS
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

- What is the NextGenXMS URL needed?
-- https://int.xms.hhs.gov/ABaaS/idp

- Does NextGenXMS have a logo we need to incorporate?
-- Yes they have a logo, but it's not required.

- Does XMS have to be explicitly mentioned on the button?
-- No. The reason to mention it would be that XMS eventually plans to have other means of logging in besides login.gov. We will change the label on the button if/when that becomes necessary.

### PR DESCRIPTION
**Pull request by riatzukiza**
## Summary of Changes

Makes the button configurable through the env
Pull request closes #1560 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```
### login.gov

1. Open http://localhost:3000/ and sign in.
2. Click login xms button
3. see login page

### XMS

1. Add `REACT_APP_LOGIN_XMS_URL=https://int.xms.hhs.gov/ABaaS/idp` to .env.development file

1. Open http://localhost:3000/ and sign in.
2. Click login xms button
3. see login page
## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [x] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by abottoms-coder — APPROVED** [review]

**Comment by abottoms-coder** [comment]
ACs are met, we need to adjust the frontend tests though.

**Comment by reitermb** [comment]
Having some trouble getting this one to build cc @riatzukiza

**Comment by jorgegonzalez** [comment]
This PR won't deploy for a11y review because there's a failing frontend test

File: tdrs-frontend/src/components/SplashPage/SplashPage.jsx [hunk 1]
```diff
@@ -28,6 +28,9 @@ function SplashPage() {
       // This doesn't need to be tested, it will never be reached by jest.
       event.preventDefault()
       dispatch(setMockLoginState())
+    } else if (process.env.REACT_APP_LOGIN_XMS_URL) {
+      event.preventDefault()
+      window.location.href = process.env.REACT_APP_LOGIN_XMS_URL
```
Comment (abottoms-coder) [inline tdrs-frontend/src/components/SplashPage/SplashPage.jsx:33]:
```suggestion
      window.location.href = `${process.env.REACT_APP_BACKEND_URL}/login/xms'

```

File: tdrs-frontend/.env [hunk 2]
```diff
@@ -45,3 +45,5 @@ INLINE_RUNTIME_CHUNK=false
 
 # Ignore CRA package version check until version is updated to 5.0.0
 SKIP_PREFLIGHT_CHECK=true
+
+REACT_APP_LOGIN_XMS_URL=https://int.xms.hhs.gov/ABaaS/idp
```
Comment (abottoms-coder) [inline tdrs-frontend/.env:49]: After trying to use this with @raftmsohani and @jorgegonzalez on an XMS branch, we have decided on the below edits.  Can this be renamed and made into a simple boolean?

```suggestion
LOGIN_XMS_USED=True
```

File: tdrs-frontend/src/components/SplashPage/SplashPage.jsx (hunk 1 above)
Comment (abottoms-coder) [inline tdrs-frontend/src/components/SplashPage/SplashPage.jsx:33]:
```suggestion
```suggestion
    } else if (process.env.LOGIN_XMS_USED) {
      event.preventDefault()
      window.location.href = `${process.env.REACT_APP_BACKEND_URL}/login/xms`
```
```

Comment (abottoms-coder) [inline tdrs-frontend/src/components/SplashPage/SplashPage.jsx:33]:
```suggestion
    } else if (process.env.LOGIN_XMS_USED) {
      event.preventDefault()
      window.location.href = `${process.env.REACT_APP_BACKEND_URL}/login/xms`
```

File: tdrs-frontend/.env (hunk 2 above)
Comment (jorgegonzalez) [reply tdrs-frontend/.env:49]: This should be lowercase in JavaScript @abottoms-coder

**Comment by reitermb** [comment]
No rush, just wanted to follow up re: the failing tests

File: tdrs-frontend/src/components/SplashPage/SplashPage.jsx (hunk 1 above)
Comment (abottoms-coder) [reply tdrs-frontend/src/components/SplashPage/SplashPage.jsx:33]: Ok, I have updated my suggestions.

**Review by reitermb — APPROVED** [review]
Looks good! Changes to button behavior here all look good, nothing impacting a11y (other than stuff outside the scope of this ticket like XMS not yet being spun up)
🚀

File: tdrs-frontend/src/components/SplashPage/SplashPage.test.js [hunk 3]
```diff
@@ -39,13 +39,22 @@ const initialState = {
 const mockStore = configureStore([thunk])
 
 describe('SplashPage', () => {
+  const OLD_ENV = process.env
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/SplashPage/SplashPage.test.js:42]: need to mock the env so we can test the behavior in either state

**Review by abottoms-coder — APPROVED** [review]
Works as expected by itself, also used this in #1561 successfully.

**Comment by jorgegonzalez** [comment]
What is the holdup on this PR again? @ADPennington @abottoms-coder

**Comment by 4l2l3** [comment]
> What is the holdup on this PR again? @ADPennington @abottoms-coder

I approved, the hold up *was* a11y-review but looks like this might be ready for a tabletop (which should include Miles). @riatzukiza Is that correct? If so, please schedule a tabletop and we can get this moved to QASP.

**Comment by abottoms-coder** [comment]
> What is the holdup on this PR again? @ADPennington @abottoms-coder

I approved, the hold up was a11y-review but looks like this might be ready for a tabletop (which should include Miles). @riatzukiza Is that correct? If so, please schedule a tabletop and we can get this moved to QASP.

**Review by jorgegonzalez — APPROVED** [review]

**Review by ADPennington — APPROVED** [review]
lgtm @riatzukiza 👍🏾 

login button redirects here in staging: `https://idp.int.identitysandbox.gov/...`

and here upon deploy to qasp: `https://tdp-backend-qasp.app.cloud.gov/v1/login/xms`

**Pull request merged**